packages/kernel-platforms/src/factory.ts

```diff
@@ -0,0 +1,94 @@
+import { assert } from '@metamask/superstruct';
+import type { Infer } from '@metamask/superstruct';
+
+import { platformConfigStruct } from './capabilities/index.ts';
+import type {
+  PlatformFactory,
+  PlatformConfig,
+  Platform,
+  Capability,
+  CapabilityName,
+  CapabilityFactories,
+  PlatformOptions,
+} from './types.ts';
+
+const validatePlatformConfig = (
+  config: Infer<typeof platformConfigStruct>,
+  known: CapabilityName[],
+): void => {
+  const configured = Object.keys(config) as CapabilityName[];
+  if (configured.some((name) => !known.includes(name))) {
+    throw new Error(
+      `Config provided entry for unregistered capability: ${configured.find((name) => !known.includes(name))}`,
+      { cause: { configured, known } },
+    );
+  }
+  // Validate the config structure - we need to cast to the full config type for validation
+  assert(config as PlatformConfig, platformConfigStruct);
```

Comment:
```suggestion
  assert(config, platformConfigStruct);
```

minor and non blocking but the casting seems unnecessary

Reply:
[af19bb3](https://github.com/MetaMask/ocap-kernel/pull/610/commits/af19bb35f674503b9790a1e0c6a722d1ae2a1406)